blog/_posts/2015-12-2-includeos-is-now-open-source.md

```diff
@@ -0,0 +1,38 @@
+
+Think about the original reasons for creating Java; one main reason was definitely portability. With pervasive hardware virtualization you can now get similar portability for compiled C++ programs. We're currently only portable across x86 operating systems, but with virtualbox, you can run the same binary on windows, Linux and Mac. What's more; "Hello World" in IncludeOS on KVM, will have a much smaller memory footprint than "Hello world" in Java, running directly on the server - even when we take the whole Qemu process into account.
+
+From this interesting premise, rather than working towards running a Java Virtual Machine, or supporting another high-level language, we want to focus on creating an intuitive and modern C++ hardware API, with minimal overhead. But of course - if somebody wants to use IncludeOS as a platform for supporting high level language runtimes, that would be great too. Before that can happen though, we'll have to get a stable API -  we're currently on 0.7.0, meaning that anything can change at any time.
+
+## We want to make Node++
+IncludeOS is pretty similar to Node.js in a few respects: It's single threaded at the moment, but highly efficient due to non-blocking I/O. This means that you'll have to use a callback-based programming style, just like with Javascript. The events you'll be subscribing to are all rooted in a physical interrupt - in fact, you can add callbacks to interrupts directly, if you want.  An interesting thing to keep in mind is that I/O in modern hardware is non-blocking by default, with Direct Memory Access (DMA). A network packet arriving into IncludeOS will result in an interrupt, but only after the whole packet is written directly into memory by the device. At that point, IncludeOS will defer the interrupt, and as soon as the CPU is done with whatever it was doing, it will call the callback delegate (possibly a lambda) of whoever's subscribing to that interrupt. That delegate can again use the same technique, to fire events higher up the stack. In our view, this kind of event-based programming fits hardware better than sequential, blocking programming. Also, GUI-programming has always been like this, so the techniques are well known. 
+
+### We're making a github-based package manager
+One very important feature we'll be needing in order to make IncludeOS grow, is a package manager. We're making one much like [npm](https://www.npmjs.com/), where anyone can create a package as a github repository, and then register that package in a central location for others to use. The idea is to have `$ include http` on the terminal, result in the possibility of writing `#include <http>` inside your IncludeOS service. This will obviously take some time, but hopefully we have something working by summer 2016.
```

Comment:
I wonder how much should be said about this.  Mentioning it as a sentence is fine but since this is the first post, I think it's easier for the reader if most of the content sticks to the things that exist today.  Later posts can elaborate on the longer term plans.  I think this will make the post easier to digest (most readers won't have heard of IncludeOS, yet).